### ISSUE TITLE
CMake: investigate warnings

### ISSUE THREAD

**Issue by maxteufel** [SuperTux/supertux#841]
Upon recompiling on Arch Linux with CMake v3.11.4, the following two warnings popped up:

```
CMake Deprecation Warning at CMakeLists.txt:49 (CMAKE_POLICY):
  The OLD behavior for policy CMP0043 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.
```

```
CMake Warning (dev) at /usr/share/cmake-3.11/Modules/FindOpenGL.cmake:270 (message):
  Policy CMP0072 is not set: FindOpenGL prefers GLVND by default when
  available.  Run "cmake --help-policy CMP0072" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  FindOpenGL found both a legacy GL library:

    OPENGL_gl_LIBRARY: /usr/lib/libGL.so

  and GLVND libraries for OpenGL and GLX:

    OPENGL_opengl_LIBRARY: /usr/lib/libOpenGL.so
    OPENGL_glx_LIBRARY: /usr/lib/libGLX.so

  OpenGL_GL_PREFERENCE has not been set to "GLVND" or "LEGACY", so for
  compatibility with CMake 3.10 and below the legacy GL library will be used.
Call Stack (most recent call first):
  CMakeLists.txt:148 (FIND_PACKAGE)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

These warnings need to be investigated and a proper fix needs to be found and implemented.

**Comment by AshishBhattarai** [comment]
`CMAKE_POLICY(SET CMP0023 NEW)` fixes policy warning but the target_link_libraries must be either
keywords or plain signatures. 

According to cmake docs.
The OLD behavior for this policy is to allow keyword and plain target_link_libraries signatures to be mixed. The NEW behavior for this policy is to not to allow mixing of the keyword and plain signatures.

**Comment by tobbi** [comment]
For the remaining warning, please see:
https://cmake.org/cmake/help/v3.0/policy/CMP0043.html

**Comment by maxteufel** [comment]
Fixed by ccd5a60037b1d930aaf26e7b4254b123b091f171.
